iocextractor/core/server.py

```diff
@@ -270,6 +275,7 @@ def __call__(self, app, *args, **kwargs):
         '''
         create_topic_if_not_exists(KAFKA_SERVER, IOC_TOPIC_NAME)
         scheduler.start()
+        Extractor.refetch_blacklist()
```

Comment:
See line 280

Reply:
Removed the duplicate.

Reply:
Check for an not empty blacklist before starting the thread.

Reply:
I am doing so inside the refetch_blacklist on line 113. But I could additionally build this logic in the __call__ function if you want.

Reply:
Does this do the job?

Reply:
Should be working. Before starting the thread or the scheduler, we are stepping into the refetch_blacklist / refetch_datasources. If the Blacklist/ the Datasources are empty, we are getting the data locally. Else we are using gitlab to get the data.